test/middleware.test.ts

```diff
@@ -1,110 +1,119 @@
-import Pyroscope, { expressMiddleware } from '@pyroscope/nodejs';
-import request from 'supertest';
-import express, { Response } from 'express';
-
+import Pyroscope, { expressMiddleware } from '../src'
+import request from 'supertest'
+import express from 'express'
 
 // You only need appName for the pull mode
-Pyroscope.init();
-
+Pyroscope.init()
 
 describe('express middleware', () => {
-    afterAll(async () => {
-        await new Promise<void>(resolve => setTimeout(() => resolve(), 500)); // avoid jest open handle error
-    });
-    it('should be a function', () => {
-        expect(typeof expressMiddleware).toBe('function');
-        
-    })
-    it('should respond to cpu calls', () => {
-        const app = express();
-        app.use(expressMiddleware());
-        return request(app).get('/debug/pprof/profile?seconds=1')
-                .then(result => expect(result.statusCode).toBe(200))
-                .catch(result => {
-                    expect(result.statusCode).toBe(200)
-                })
-    })
-    it('should respond to repetitive cpu calls', () => {
-        const app = express();
-        app.use(expressMiddleware());
-        return request(app).get('/debug/pprof/profile?seconds=1')
-                .then(result => expect(result.statusCode).toBe(200))
-                .catch(result => {
-                    expect(result.statusCode).toBe(200)
-                })
-    })
+  afterAll(async () => {
+    await new Promise<void>((resolve) => setTimeout(() => resolve(), 500)) // avoid jest open handle error
+  })
+  it('should be a function', () => {
+    expect(typeof expressMiddleware).toBe('function')
+  })
+  it('should respond to cpu calls', async () => {
+    const app = express()
+    app.use(expressMiddleware())
+    return request(app)
+      .get('/debug/pprof/profile?seconds=1')
+      .then((result) => {
+        expect(result.statusCode).toBe(200)
+      })
+      .catch((result) => {
+        expect(result.statusCode).toBe(200)
+      })
+  })
+  it('should respond to repetitive cpu calls', async () => {
+    const app = express()
```

Comment:
@korniltsev we need to think about this case. What should we do on consecutive calls? How do we handle the profiler? The old implementation closed the profiler so it could not handle consecutive cpu calls